exporter/coralogixexporter/client.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coralogixexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/coralogixexporter"
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"os"
+
+	cxpb "github.com/coralogix/opentelemetry-cx-protobuf-api/coralogixpb"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+)
+
+// CoralogixClient by Coralogix
+type CoralogixClient struct {
```

Comment:
Should this be exported?

Reply:
No, I will change this. the access to coralogixClient is internally, only from coralogixexporter package

Reply:
but after I changed this to "coralogixClient", the linter throw exception on function NewCoralogixClient declaration in client.go.
`unexported-return: exported func NewCoralogixClient returns unexported type *coralogixexporter.coralogixClient, which can be annoying to use (revive)`.
I can't declare this function with return value "*coralogixClient"

Reply:
Because `NewCoralogixClient` also likely needs to kept private.

Reply:
Thanks, I will change this